docs/reference/extensions/gnome-extension.rst

```diff
@@ -0,0 +1,253 @@
+----------------
+
+This extension uses `layouts <https://snapcraft.io/docs/snap-layouts>`_ to access files
+on the host. The platform snap's gjs, webkit2gtk-4.0, and iso-codes are used so they
+don't need to be packaged as part of the snap and would greatly inflate the size.
+
+.. tabs::
+
+    .. group-tab:: core24
+
+        .. collapse:: Included layouts
+
+            .. code-block:: yaml
+
+                layout:
+                  /usr/lib/$CRAFT_ARCH_TRIPLET_BUILD_FOR/webkit2gtk-4.0:
+                    bind: $SNAP/gnome-platform/usr/lib/$CRAFT_ARCH_TRIPLET_BUILD_FOR/webkit2gtk-4.0
+                  /usr/lib/$CRAFT_ARCH_TRIPLET_BUILD_FOR/webkit2gtk-4.1:
+                    bind: $SNAP/gnome-platform/usr/lib/$CRAFT_ARCH_TRIPLET_BUILD_FOR/webkit2gtk-4.1
+                  /usr/share/xml/iso-codes:
+                    bind: $SNAP/gnome-platform/usr/share/xml/iso-codes
+                  /usr/share/libdrm:
+                    bind: $SNAP/gpu-2404/libdrm
+                  /usr/share/drirc.d:
+                    symlink: $SNAP/gpu-2404/drirc.d
+                  /usr/share/X11/XErrorDB:
+                    symlink: $SNAP/gpu-2404/X11/XErrorDB
+
+    .. group-tab:: core22
+
+        .. collapse:: Included layouts
+
+            .. code-block:: yaml
+
+                layout:
+                  /usr/lib/$SNAPCRAFT_ARCH_TRIPLET/libgweather-4:
+                    symlink: $SNAP/usr/lib/$SNAPCRAFT_ARCH_TRIPLET/libgweather-4
+                  /usr/lib/evolution-data-server:
+                    symlink: $SNAP/usr/lib/evolution-data-server
+                  /usr/bin/gnome-control-center:
+                    symlink: $SNAP/usr/bin/gnome-control-center
+                  /usr/lib/$SNAPCRAFT_ARCH_TRIPLET/webkit2gtk-4.0:
+                    bind: $SNAP/gnome-platform/usr/lib/$SNAPCRAFT_ARCH_TRIPLET/webkit2gtk-4.0
+                  /usr/share/xml/iso-codes:
+                    bind: $SNAP/gnome-platform/usr/share/xml/iso-codes
+                  /usr/share/libdrm:
+                    bind: $SNAP/gnome-platform/usr/share/libdrm
+
+
+Example expanded project file
+-----------------------------
+
+Here's an example of the result of Snapcraft expanding a core24-based project file, as
+immediately prior to build. It demonstrates the added plugs, packages, variables, and
+layouts, that the GNOME extension includes in a project.
+
+The original file was the `GNOME System Monitor snap
+<https://snapcraft.io/gnome-system-monitor>`_. This text contains the difference between
+the original file and the output of the :ref:`snapcraft expand-extensions
+<ref_commands_expand-extensions>` command. Some of the text has been altered for ease of
+reading.
+
+.. collapse:: Expanded project file for GNOME System Monitor
+
+    .. literalinclude:: ../code/extensions/gnome-extension-gnome-system-monitor-expanded-diff.yaml
+        :language: diff
+        :lines: 3-
+        :emphasize-lines: 57-66, 78-87, 100-150, 158-163, 171-173, 180-201, 210-212
```

Comment:
I'd like to hear initial impressions on this. There are several combinations of visual presentations we could try, but I thought for the first one I'd turn everything on.

Reply:
I like this. Ideally we'd have a way to emphasise any line that starts with a `+` character instead, but I think that's beyond the scope of what sphinx does.

Reply:
Glad it works for you.

As far as automagically detecting `+`, manual selection has at least one advantage right now. There are several kinds of code changes from `expand-extensions` that are stylistic and mechanical (like the automatic linter exceptions) and have no real bearing on understanding extensions.